sklearn/impute.py

```diff
@@ -917,7 +936,19 @@ def fit_transform(self, X, y=None):
             if self.verbose > 0:
                 print('[IterativeImputer] Ending imputation round '
                       '%d/%d, elapsed time %0.2f'
-                      % (i_rnd + 1, self.n_iter, time() - start_t))
+                      % (i_rnd + 1, self.max_iter, time() - start_t))
+
+            # stop early if difference between consecutive imputations goes up.
+            # if so, back off to previous imputation
+            if not self.sample_posterior:
+                if np.max(np.abs(Xt - Xt_previous)) < self.tol:
+                    self.n_iter_ = i_rnd + 1
+                    if self.verbose > 0:
+                        print('[IterativeImputer] Early stopping criterion '
+                              'reached.')
+                    break
+                else:
```

Comment:
you can remove the `else:`

Reply:
I don't think that's true. I need to specify that `Xt_previous = Xt.copy()` otherwise the subsequent diff will be 0. Try it out.

Reply:
what I mean is 

```python
            if not self.sample_posterior:
                inf_norm = np.linalg.norm(Xt - Xt_previous, ord=np.inf,
                                          axis=None)
                if inf_norm < normalized_tol:
                    self.n_iter_ = i_rnd + 1
                    if self.verbose > 0:
                        print('[IterativeImputer] Early stopping criterion '
                              'reached.')
                    break
                Xt_previous = Xt.copy()
```

You don't need else since you will break in the if branch

Reply:
Oh! I thought you meant that the entire else branch was not needed. Yup, you're right. I'll fix it.